File: arm.64/langs/swift/src/boot.swift
import Foundation

func boot(_ targetDir: String? = nil) -> [Cabi] {
    var engines: [Cabi] = []
    if let envLib = ProcessInfo.processInfo.environment["PAT_LIB"], !envLib.isEmpty {
        if let c = bind(envLib) {
            engines.append(c)
            return engines
        }
    }

    let files = find(targetDir)
    for file in files {
        if let c = bind(file) {
            engines.append(c)
        }
    }
    return engines
}

File: arm.64/langs/rust/src/boot.rs
use crate::bind::bind;
use crate::find::find;
use crate::r#type::Cabi;
use std::env;

pub fn boot(target_dir: Option<&str>) -> Vec<Cabi> {
    let mut engines = Vec::new();
    if let Ok(env_lib) = env::var("PAT_LIB") {
        if !env_lib.is_empty() {
            if let Some(c) = bind(&env_lib) {
                engines.push(c);
                return engines;
            }
        }
    }

    let files = find(target_dir);
    for file in files {
        if let Some(c) = bind(&file) {
            engines.push(c);
        }
    }
    engines
}
